k9mail-library/src/test/java/com/fsck/k9/mail/filter/LineWrapOutputStreamTest.java

```diff
@@ -0,0 +1,308 @@
+package com.fsck.k9.mail.filter;
+
+import org.junit.Test;
+
+import java.io.BufferedWriter;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.UnsupportedEncodingException;
+import java.util.Random;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * Created by ployt0 on 2/15/17.
+ * Java quotes by Bruce Eckel, Thinking In Java
+ */
+public class LineWrapOutputStreamTest {
+    private ByteArrayOutputStream baos = new ByteArrayOutputStream(64000);
+    private Random rand = new Random();
```

Comment:
Unconstrained randomness in tests is unhelpful because when you re-run it and it passes you don't know whether it was really a failure in the code. And you aren't logging the random values used so there'd be no way to reproduce the error.